app/src/sharedTest/java/org/oppia/android/app/classroom/ClassroomListFragmentTest.kt

```diff
@@ -0,0 +1,886 @@
+      .assertTextContains(context.getString(R.string.recommended_stories))
+      .assertIsDisplayed()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_TEST_TAG).onChildAt(0)
+      .assertTextContains("What is a Fraction?")
+      .assertTextContains("FRACTIONS")
+      .assertTextContains("MATHS")
+      .assertIsDisplayed()
+  }
+
+  @Test
+  fun testFragment_noTopicProgress_initialRecommendationFractionsAndRatiosIsCorrect() {
+    logIntoAdminTwice()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_HEADER_TEST_TAG).onChildAt(0)
+      .assertTextContains(context.getString(R.string.recommended_stories))
+      .assertIsDisplayed()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_TEST_TAG).apply {
+      onChildAt(0)
+        .assertTextContains("What is a Fraction?")
+        .assertTextContains("FRACTIONS")
+        .assertTextContains("MATHS")
+        .assertIsDisplayed()
+
+      onChildAt(1)
+        .assertTextContains("What is a Ratio?")
+        .assertTextContains("RATIOS AND PROPORTIONAL REASONING")
+        .assertTextContains("MATHS")
+        .assertIsDisplayed()
+    }
+  }
+
+  @Test
+  fun testFragment_forPromotedActivityList_hideViewAll() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_UPTIME_MILLIS)
+    storyProgressTestHelper.markInProgressSavedFractionsStory0Exp0(
+      profileId = profileId,
+      timestampOlderThanOneWeek = false
+    )
+    logIntoAdminTwice()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_HEADER_TEST_TAG).onChildAt(1)
+      .assertDoesNotExist()
+  }
+
+  @Test
+  @Ignore("Temporarily ignored as the test is failing.")
+  // TODO(#5344): Update the logic or fix the test.
+  fun testFragment_markStory0DoneForRatiosAndFirstTestTopic_displaysRecommendedStories() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_UPTIME_MILLIS)
+    storyProgressTestHelper.markCompletedTestTopic0Story0(
+      profileId = profileId,
+      timestampOlderThanOneWeek = false
+    )
+    storyProgressTestHelper.markCompletedRatiosStory0(
+      profileId = profileId,
+      timestampOlderThanOneWeek = false
+    )
+    logIntoAdminTwice()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_HEADER_TEST_TAG).onChildAt(0)
+      .assertTextContains(context.getString(R.string.recommended_stories))
+      .assertIsDisplayed()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_TEST_TAG).onChildAt(0)
+      .assertTextContains("Fifth Exploration")
+      .assertTextContains("SECOND TEST TOPIC")
+      .assertTextContains("SCIENCE")
+      .assertIsDisplayed()
+  }
+
+  @Test
+  // TODO(#5344): Add tests for coming soon topics list.
+  fun testFragment_markAtLeastOneStoryCompletedForAllTopics_displaysComingSoonTopicsList() {
+  }
+
+  @Test
+  // TODO(#5344): Add tests for coming soon topics list.
+  fun testHomeActivity_markFullProgressForSecondTestTopic_displaysComingSoonTopicsText() {
+  }
```

Comment:
We should not check in incomplete code. You can delete these and make an external note of these.

Reply:
Removed.